### PR TITLE
Change esnext.asynciterable to es2018.asynciterable

### LINKED ISSUE #26218: esnext.asynciterable.d.ts should be es2018.asynciterable.d.ts
https://github.com/Microsoft/TypeScript/blob/9706f1729e9c41c9af901c279367dec0ab1093d1/src/lib/esnext.d.ts

```ts
/// <reference lib="es2018" />
/// <reference lib="esnext.asynciterable" />
/// <reference lib="esnext.array" />
/// <reference lib="esnext.symbol" />
/// <reference lib="esnext.intl" />
```

Currently Symbol.asyncIterator is in esnext but it should be in [es2018](https://github.com/tc39/proposals/blob/master/finished-proposals.md).

### PR DESCRIPTION
**Pull request by saschanaz**
Fixes #26218

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
Please resolve merge conflict. Thanks!

**Comment by saschanaz** [comment]
@RyanCavanaugh Done and the test passes!

**Comment by johndoe23612896381634** [comment]
When will this PR be merged?

**Comment by johndoe2361289638165** [comment]
C'mon Microsoft, pls merge and release it!

**Comment by jordanbtucker** [comment]
Async iterators were moved to Stage 4 almost a year ago. Does it normally take this long to update lib typings?

**Comment by johndoe2361289638165** [comment]
Still not merged...? C'mon...

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -38,12 +38,12 @@ namespace ts {
         ["es2017.string", "lib.es2017.string.d.ts"],
         ["es2017.intl", "lib.es2017.intl.d.ts"],
         ["es2017.typedarrays", "lib.es2017.typedarrays.d.ts"],
+        ["es2018.asynciterable", "lib.es2018.asynciterable.d.ts"],
         ["es2018.intl", "lib.es2018.intl.d.ts"],
         ["es2018.promise", "lib.es2018.promise.d.ts"],
         ["es2018.regexp", "lib.es2018.regexp.d.ts"],
         ["esnext.array", "lib.esnext.array.d.ts"],
         ["esnext.symbol", "lib.esnext.symbol.d.ts"],
-        ["esnext.asynciterable", "lib.esnext.asynciterable.d.ts"],
```
Comment (rbuckton) [inline src/compiler/commandLineParser.ts]: For backwards-compatibility sake we might want to have `lib.esnext.asynciterable.d.ts` that just has a `/// <reference lib="es2018.asynciterable" />`

Comment (rbuckton) [reply src/compiler/commandLineParser.ts]: Actually, a simpler approach would be to have `["esnext.asynciterable", "lib.es2018.asynciterable.d.ts"],` here.

Comment (saschanaz) [reply src/compiler/commandLineParser.ts]: Done!

**Comment by johndoe2361289638165** [comment]
..........

**Comment by rbuckton** [comment]
This looks good although the PR seems to have caught up some recent changes in master that seem to be unrelated, so I may have to merge manually.

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**